src/z.js

```diff
@@ -1,56 +1,62 @@
-var pos = require("ni-pos");
-var pluralize = require("pluralize");
+const pos = require("ni-pos");
+const pluralize = require("pluralize");
 
 const ACCEPTED_TAG_TYPE = ["NNS", "NNPS"];
-const SYMBOL_TAG_TYPE = [",", ".", ":", "$", "#", '"', ")", "("];
+const EMOJI_REGEXP = /:\w+:/
 
-function isAnEmoji(currentIndex, taggedWords) {
-  return (
-    taggedWords[currentIndex + 2] && 
-    taggedWords[currentIndex + 2][0] === ":"
-  );
+const lexer = new pos.Lexer();
+// Add support for Slack emoji in lexing
+lexer.regexs = [EMOJI_REGEXP, ...lexer.regexs]
```

Comment:
This makes this simpler -- turns out it's quite easy to just hack in a new regexp to the lexer to let it recognize `:thing:` as a single token. No more index juggling.